### PR TITLE
Various python

### PR DESCRIPTION
**Pull request by teh**
###### Motivation for this change

Don't merge just yet, running nox-review fails locally so I want to try the `pr` option.

```
error: while querying the derivation named ‘tex4ht-1.0.2009_06_11_1038’:
while evaluating the attribute ‘buildPhase’ of the derivation ‘tex4ht-1.0.2009_06_11_1038’ at /home/tom/.nox/nixpkgs/pkgs/tools/typesetting/tex/tex4ht/default.nix:4:3:
cannot coerce null to a string, at /home/tom/.nox/nixpkgs/pkgs/tools/typesetting/tex/tex4ht/default.nix:15:16
```

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by teh** [comment]
Still no luck, not sure what's up, could be related to 2aff29c7b202725933dd5306eed4241fe86be0b9 by @vcunat ?

In any case, I build by hand:

```
nix-build -A pythonPackages.tables
nix-build -A pythonPackages.Logbook
nix-build -A python3Packages.tables
nix-build -A python3Packages.Logbook
```

and everything succeeded

**Comment by vcunat** [comment]
Ah, yes.  It should be OK now.  Some things don't behave very smart, unfortunately.

**Comment by teh** [comment]
I get some bizarre errors for rpy2 when building on debian [1] but on nixos `nox-review pr 22873`  now succeeds after fixing unrelated failures in vega and altair.

Can't wait for PR testing by Hydra!

[1] - linking again the local readline .. bizarre, but rpy2 does a lot of weird stuff anyway due to R being weird.
```
ldd /nix/store/dsg3kdf3z1gc4ycdk74b11s51svywix9-python2.7-rpy2-2.8.2/lib/python2.7/site-packages/rpy2/rinterface/_rinterface.so  | grep /lib/x
	libreadline.so.6 => /lib/x86_64-linux-gnu/libreadline.so.6 (0x00007f21a820f000)
	libtinfo.so.5 => /lib/x86_64-linux-gnu/libtinfo.so.5 (0x00007f21a37cc000)
```

**Comment by vcunat** [comment]
You would avoid most of these problems by using nix's sandboxing &ndash; that way the builders won't see any `/usr/` or `/lib/`.

**Comment by teh** [comment]
@vcunat - thanks. My laptop runs a single-user nix packages on debian which is a slightly unusual arrangement. I'm not sure that's worth blocking on.

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -417,15 +417,16 @@ in {
...
 
     src = pkgs.fetchurl {
       url = "mirror://pypi/a/altair/${name}.tar.gz";
-      sha256 = "024drhmiw8w3dl7dbal0pvnlfd3sv4n1rqywv2jb488b3fzm704r";
+      sha256 = "05c47dm20p7m0017p2h38il721rxag1q0457dj7whp0k8rc7qd1n";
     };
-
+    # Tests try to execute notebooks that aren't tests:
+    # TypeError: calling <function tutorial at 0x7fffe1bb8048> returned None, not a test
+    doCheck = false;
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: maybe the test runner should be invoked differently?

**Comment by FRidh** [comment]
@teh Could you rebase and update this PR?

File: pkgs/top-level/python-packages.nix (hunk 1 above)
Comment (teh) [inline pkgs/top-level/python-packages.nix]: figured it out, needs a UTF locale as well, hope this translates to hydra (works locally).

**Comment by teh** [comment]
rebased, PTAL

**Review by FRidh — APPROVED** [review]

**Comment by vcunat** [comment]
I see `python3Packages.altair` fails tests, but it was so before this PR as well...

**Pull request merged**